src/lib/APS2Ethernet.cpp

```diff
@@ -0,0 +1,560 @@
+// Ethernet communications with APS2 boards
+//
+// Original authors: Colm Ryan, Blake Johnson, Brian Donovan
+// Copyright 2016, Raytheon BBN Technologies
+
+#include "APS2Ethernet.h"
+
+#ifdef _WIN32
+#include "iphlpapi.h"
+#else
+#include <ifaddrs.h>
+#endif
+
+APS2Ethernet::APS2Ethernet() : udp_socket_old_(ios_), udp_socket_(ios_) {
+    FILE_LOG(logDEBUG) << "APS2Ethernet::APS2Ethernet";
+
+    //Move the io_service to a background thread
+    //Give it something to chew on so that it doesn't return from the background thread
+    asio::io_service::work work(ios_);
+    receiveThread_ = std::thread([this](){ ios_.run(); });
+
+    //Bind the old UDP socket at local port bb4e
+    try {
+        udp_socket_old_.open(udp::v4());
+        udp_socket_old_.bind(udp::endpoint(udp::v4(), APS_PROTO));
+    } catch (...) {
+        throw APS2_SOCKET_FAILURE;
+    }
+
+    //Bind UDP socket at local port bb4f
+    try {
+        udp_socket_.open(udp::v4());
+        udp_socket_.bind(udp::endpoint(udp::v4(), APS_PROTO+1));
```

Comment:
Just to confirm: the future plan here is to not bind to any particular local socket and have the APS2 respond back to whichever socket the message came from, right?

(I understand that the firmware does not do this, yet).